tracer/tracer.go

```diff
@@ -244,16 +245,16 @@ func (t *Tracer) NewChildSpanWithContext(name string, ctx context.Context) (*Spa
 
 // SetDebugLogging will set the debug level
 func (t *Tracer) SetDebugLogging(debug bool) {
-	t.debugMu.Lock()
-	defer t.debugMu.Unlock()
-	t.debugLoggingEnabled = debug
+	if debug {
+		atomic.CompareAndSwapUint32(&t.debugMode, 0, 1)
+	} else {
+		atomic.CompareAndSwapUint32(&t.debugMode, 1, 0)
+	}
 }
 
 // DebugLoggingEnabled returns true if the debug level is enabled and false otherwise.
 func (t *Tracer) DebugLoggingEnabled() bool {
-	t.debugMu.RLock()
-	defer t.debugMu.RUnlock()
-	return t.debugLoggingEnabled
+	return t.debugMode == 1
```

Comment:
while reading the value, you shouldn't use the `LoadUint32`? so:
```go
return atomic.LoadUint32(&t.debugMode) == 1
```

Reply:
You're right. Done.